View Book Receive

esterinac opened the pull request:
- Ada 2 progress di book receive yaitu handover (serah terima) dan wrapping
- Tabel book_receive_user digunakan untuk mengisi data staff yang bertugas tiap progress di book receive
- Contoh data book_receive_user
![image](https://user-images.githubusercontent.com/66486312/115484016-27a98e80-a27c-11eb-8ad4-c4e4657851fa.png)

- Screenshot
![screencapture-localhost-sigap-book-receive-view-1-2021-04-21-08_29_22](https://user-images.githubusercontent.com/66486312/115484121-62abc200-a27c-11eb-9653-426f3280f787.png)
   - deadline_modal (deadline per progress):
![image](https://user-images.githubusercontent.com/66486312/115484297-c209d200-a27c-11eb-8ddf-9fff3a8260c2.png)
   - input_staff (isi staff per progress)
![image](https://user-images.githubusercontent.com/66486312/115484337-d9e15600-a27c-11eb-8bc1-6fe55aa8d224.png)
   - action_modal (untuk approval)
![image](https://user-images.githubusercontent.com/66486312/115484410-f8475180-a27c-11eb-9bc7-ca1b6976c7ea.png)
   - notes_modal (catatan per progress)
![image](https://user-images.githubusercontent.com/66486312/115484441-06956d80-a27d-11eb-9eb6-ca0063d69a10.png)
   - final (finalisasi penerimaan buku)
![image](https://user-images.githubusercontent.com/66486312/115484586-3f354700-a27d-11eb-9928-25b668ba0bae.png)

esterinac committed:
Create table book_receive_user

esterinac committed:
View book receive

- Add deadline book receive
- Progress Bar
- Start, finish, action progress (handover & wrapping)
- Select staff for progress
- Final book receive

esterinac committed:
Generate pdf handover (serah terima) & wrapping

bagaskarala commented:
staff gudang belum ada di pilihan user? @esterinac

bagaskarala commented:
file berita acara tidak disimpan di database?
aku menemukan case ketika upload berita acara kemudian klik selesai, maka tombol download hilang. file masih ada tetap tidak ngelink ke data manapun.

bagaskarala commented:
perlu ditambahkan keterangan ketika user diminta mengisikan deadline terlebih dahulu

bagaskarala commented:
oiya btw data stock book bisa dihapus? nah create-nya darimana?

esterinac commented:
> staff gudang belum ada di pilihan user? @esterinac

belum mas kemarin aku cuma insert sendiri lewat phpmyadmin. berarti ditambahin pilihan level staff gudang ya mas pas nambahin akun user di sini 
![image](https://user-images.githubusercontent.com/66486312/116225846-7ffcf680-a77c-11eb-80ba-1396a4999965.png)

esterinac commented:
> perlu ditambahkan keterangan ketika user diminta mengisikan deadline terlebih dahulu

maksudnya keterangan seperti 'Isi deadline terlebih dahulu sebelum memulai ...' gitu ya mas?

esterinac commented:
> oiya btw data stock book bisa dihapus? nah create-nya darimana?

iya mas bisa dihapus. sementara ini data book stock cuma bisa ditambahin saat ada penerimaan buku baru mas. kemarin lupa bikin create-nya, kira-kira lebih bagus kalo dibikin create-nya ya mas?

esterinac committed:
Add staff gudang user

esterinac committed:
Keterangan isi deadline

bagaskarala commented:
@esterinac Yap betul

bagaskarala commented:
> > perlu ditambahkan keterangan ketika user diminta mengisikan deadline terlebih dahulu
> 
> maksudnya keterangan seperti 'Isi deadline terlebih dahulu sebelum memulai ...' gitu ya mas?

Iya untuk petunjuk user

bagaskarala commented:
> > oiya btw data stock book bisa dihapus? nah create-nya darimana?
> 
> iya mas bisa dihapus. sementara ini data book stock cuma bisa ditambahin saat ada penerimaan buku baru mas. kemarin lupa bikin create-nya, kira-kira lebih bagus kalo dibikin create-nya ya mas?

Nah data book stock relasinya gimana dg book dan book receive?

Kalo data book stock cuma ada satu per buku,, brrti ga perlu ada create dan delete, karena ngikut dari bukunya

esterinac commented:
> > > oiya btw data stock book bisa dihapus? nah create-nya darimana?
> > 
> > 
> > iya mas bisa dihapus. sementara ini data book stock cuma bisa ditambahin saat ada penerimaan buku baru mas. kemarin lupa bikin create-nya, kira-kira lebih bagus kalo dibikin create-nya ya mas?
> 
> Nah data book stock relasinya gimana dg book dan book receive?
> 
> Kalo data book stock cuma ada satu per buku,, brrti ga perlu ada create dan delete, karena ngikut dari bukunya

book stock dengan book relasinya one to one mas, kalo book dengan book receive relasinya one to many.

kalau create-nya kepake buat pas buku yg baru dibuat blm pernah dicatet stock nya mas, kalau delete untuk misal kalau data bukunya udah dihapus

ichsanna committed:
Fix handover download filename

ichsanna commented:
> file berita acara tidak disimpan di database?
> aku menemukan case ketika upload berita acara kemudian klik selesai, maka tombol download hilang. file masih ada tetap tidak ngelink ke data manapun.

Untuk file berita acara tidak disimpan di database Mas, soalnya nama file sudah bisa diketahui dari nomor order. Untuk file upload dan downloadnya seharusnya sekarang sudah bisa Mas, yang sebelumnya namanya pakai entry date yg berubah ketika sudah disubmit, sekarang diganti jadi pakai nomor order.

ichsanna commented:
> > > oiya btw data stock book bisa dihapus? nah create-nya darimana?
> > 
> > 
> > iya mas bisa dihapus. sementara ini data book stock cuma bisa ditambahin saat ada penerimaan buku baru mas. kemarin lupa bikin create-nya, kira-kira lebih bagus kalo dibikin create-nya ya mas?
> 
> Nah data book stock relasinya gimana dg book dan book receive?
> 
> Kalo data book stock cuma ada satu per buku,, brrti ga perlu ada create dan delete, karena ngikut dari bukunya

Kalau book stock tidak pakai delete, berarti dibuat delete book stock saat delete book di controller book saja atau bagaimana ya Mas?

bagaskarala commented:
> > > > oiya btw data stock book bisa dihapus? nah create-nya darimana?
> > > 
> > > 
> > > iya mas bisa dihapus. sementara ini data book stock cuma bisa ditambahin saat ada penerimaan buku baru mas. kemarin lupa bikin create-nya, kira-kira lebih bagus kalo dibikin create-nya ya mas?
> > 
> > 
> > Nah data book stock relasinya gimana dg book dan book receive?
> > Kalo data book stock cuma ada satu per buku,, brrti ga perlu ada create dan delete, karena ngikut dari bukunya
> 
> book stock dengan book relasinya one to one mas, kalo book dengan book receive relasinya one to many.
> 
> kalau create-nya kepake buat pas buku yg baru dibuat blm pernah dicatet stock nya mas, kalau delete untuk misal kalau data bukunya udah dihapus

kalo book dengan book receive relasinya one to many. <-- ini contohnya gimana?

bagaskarala commented:
> Kalau book stock tidak pakai delete, berarti dibuat delete book stock saat delete book di controller book saja atau bagaimana ya Mas?

nah kalau memang relasinya begitu, berarti ya create dan delete-nya ngikut dari yang lain.
contohnya kayak di lembar-kerja/worksheet, dia gak ada create dan delete karena relasi 1-1 dengan draft, jadi ngikut data draft.

kalau contoh kasusnya gimana ya? misal ada buku **Akuntansi Pengantar** yang dicetak, lalu kemungkinan penerimaan buku dan stok buku gmn?
@esterinac  @ichsanna

esterinac commented:
> > > > > oiya btw data stock book bisa dihapus? nah create-nya darimana?
> > > > 
> > > > 
> > > > iya mas bisa dihapus. sementara ini data book stock cuma bisa ditambahin saat ada penerimaan buku baru mas. kemarin lupa bikin create-nya, kira-kira lebih bagus kalo dibikin create-nya ya mas?
> > > 
> > > 
> > > Nah data book stock relasinya gimana dg book dan book receive?
> > > Kalo data book stock cuma ada satu per buku,, brrti ga perlu ada create dan delete, karena ngikut dari bukunya
> > 
> > 
> > book stock dengan book relasinya one to one mas, kalo book dengan book receive relasinya one to many.
> > kalau create-nya kepake buat pas buku yg baru dibuat blm pernah dicatet stock nya mas, kalau delete untuk misal kalau data bukunya udah dihapus
> 
> kalo book dengan book receive relasinya one to many. <-- ini contohnya gimana?

contohnya misal tiap data book receive punya book_id yang menandakan penerimaan suatu buku. Satu buku kan bisa dicetak berkali-kali, jadi di data book receive, book_id nya bisa berulang. Tapi kalau book receive dengan print_order one to one karena penerimaan buku masuk gudang itu mengikuti print order mas.

esterinac commented:
> > Kalau book stock tidak pakai delete, berarti dibuat delete book stock saat delete book di controller book saja atau bagaimana ya Mas?
> 
> nah kalau memang relasinya begitu, berarti ya create dan delete-nya ngikut dari yang lain.
> contohnya kayak di lembar-kerja/worksheet, dia gak ada create dan delete karena relasi 1-1 dengan draft, jadi ngikut data draft.
> 

> kalau contoh kasusnya gimana ya? misal ada buku **Akuntansi Pengantar** yang dicetak, lalu kemungkinan penerimaan buku dan stok buku gmn?
> @esterinac @ichsanna

baik mas berarti yang create dan delete tidak usah dibikin bisa di-create dan delete, langsung ngikut data bukunya.

misal ada buku **Akuntansi Pengantar** yang dicetak. pertama data print_order untuk buku **Akuntansi Pengantar** akan masuk ke dalam tabel book_receive, dan selanjutnya pengguna harus mengisi deadline penerimaan buku dan menyelesaikan progress-progress yang ada di penerimaan buku yaitu serah terima dan wrapping buku. 
![image](https://user-images.githubusercontent.com/66486312/116496586-99fb1e00-a8cf-11eb-992a-40eda03ac56a.png)

setelah menyelesaikan progress-progress tersebut, akan ada finalisasi penerimaan buku. stok buku **Akuntansi Pengantar** yang ada di gudang akan ditambahkan pada saat submit finalisasi penerimaan buku. 
![image](https://user-images.githubusercontent.com/66486312/116496803-13930c00-a8d0-11eb-9dee-51a10aec334c.png)
Jika buku **Akuntansi Pengantar** belum pernah ada di stok buku maka data buku tersebut akan diinsert ke dalam book_stock dengan total stok gudang sama dengan total buku yang terjilid saat cetak, tapi kalau sebelumnya data buku **Akuntansi Pengantar** sudah ada di stok buku maka jumlah buku di gudang hanya akan diupdate dengan melakukan penambahan stok gudang yang sebelumnya sudah ada dengan total buku yang terjilid.

bagaskarala commented:
oke oke nice, berarti jadinya dihilangkan ini tombol deletenya? karena tidak ada tombol createnya.
oiya sama dirapikan bagian filternya agar full.
![image](https://user-images.githubusercontent.com/14112952/116564481-39033280-a92f-11eb-956d-5957a2b95576.png)
![image](https://user-images.githubusercontent.com/14112952/116564519-415b6d80-a92f-11eb-8c2b-a2a33ad58514.png)

gustikadek committed:
Merapikan tampilan stok buku dan penerimaan buku

1. Hapus delete
2. Delete book stock saat mendelete buku
3. Merapikan filter book stock dan penerimaan buku

gustikadek commented:
> 
> 
> oke oke nice, berarti jadinya dihilangkan ini tombol deletenya? karena tidak ada tombol createnya.
> oiya sama dirapikan bagian filternya agar full.
> ![image](https://user-images.githubusercontent.com/14112952/116564481-39033280-a92f-11eb-956d-5957a2b95576.png)
> ![image](https://user-images.githubusercontent.com/14112952/116564519-415b6d80-a92f-11eb-8c2b-a2a33ad58514.png)

Baik mas, sudah kami rapikan dan kami hilangkan button delete-nya
